src/core/Components/index.ts

```diff
@@ -175,7 +175,7 @@ export class Components implements Disposable {
     const disposer = this.tools.get(Disposer);
     this.enabled = false;
     await this.tools.dispose();
-    await this.ui.dispose();
+    await this.ui?.dispose();
```

Comment:
@HoyosJuan It will still throw, right? `this.ui` gets evaluated no matter what

```ts
  get ui(): UIManager {
    if (!this._ui) {
      throw new Error("UIManager hasn't been initialised.");
    }
    return this._ui;
  }
```

Reply:
Ah 😅 you're right! So sorry, I forget to use the private variable, not the getter.

```suggestion
    await this._ui?.dispose();
```

Reply:
It should be good now thanks to PR https://github.com/IFCjs/components/pull/271. Thanks @janruo!